src/public class Main {.java

```diff
@@ -0,0 +1,105 @@
+
+import components.set.Map;
+import components.set.Map1L;
+import components.simplereader.SimpleReader;
+import components.simplereader.SimpleReader1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+import components.naturalnumber.NaturalNumber;
+import components.naturalnumber.NaturalNumberSecondary;
+
+/**
+
+ *
+ * @author Shachi Mahajan
+ *
+ */
+public class SalesTracker extends Map1L {
+
+    /*
+     * Private members --------------------------------------------------------
+     */
+
+    /**
+     * Representation of {@code this}.
+     */
+    private Map<String, int> rep;
+
+    /**
+     * Creator of initial representation.
+     */
+    private void createNewRep() {
+
+        this.rep = new Map1L<String, int>
```

Comment:
Generic types usually have to be classes. Recent versions of Java might allow primitive types now.